Various improvements

Mat2095 opened the pull request.

Mat2095 committed:
Various improvements

RowanHarley commented:
This looks similar to the branch beta-versions. You musn't of seen that branch.

Mat2095 commented:
I have seen that. There are several differences:
- removed global lastmsg and thismsg
- use correct x and y coordinatees
- disable movement-packet since it doesn't work correctly
- move coordinate-conversion to a better place
- fix edge case (in the first packet lastTime is still unset)
- actually implement the broadcast function
- use correct snake speed
- send the correct snake speed, not some dummy-value (which was wrong anyway)

Mat2095 committed:
Even more improvements

- fix snake-packet size, speed-, coordinates-, head-coordinates- and
body-coordinates-conversion
- remove useless round
